modules/homepage/src/main/frontend/src/themePage/Navbars/SearchBar.jsx

```diff
@@ -120,12 +120,13 @@ function SearchBar(props) {
   // * the form's subject name (if available) or the resource's uuid
   function QuickSearchResultHeader(props) {
     const {resultData} = props;
+    const entry = /Forms\/(.+)/.exec(resultData["@path"]);
     return resultData && (
       <div>
         <Typography variant="body2" color="textSecondary">
           {(resultData.questionnaire?.title?.concat(' ') || '') + (resultData["jcr:primaryType"]?.replace(/lfs:/,"") || '')}
         </Typography>
-        {resultData.subject?.identifier || resultData["jcr:uuid"] || ''}
+        {resultData.subject?.identifier || entry?.[1] || ''}
```

Comment:
With the new `@name` that needs to be added:

```suggestion
        {resultData.subject?.identifier || resultData["@name"] || ''}
```